doc/source/data/quickstart.rst

```diff
@@ -3,21 +3,52 @@
 Quickstart
 ==========
 
-Learn about :class:`Dataset <ray.data.Dataset>` and the capabilities it provides.
+This page introduces the :class:`Dataset <ray.data.Dataset>` concept and the capabilities it provides.
 
 This guide provides a lightweight introduction to:
 
+* :ref:`Key Concepts: Datasets and Blocks <data_key_concepts>`
 * :ref:`Loading data <loading_key_concept>`
 * :ref:`Transforming data <transforming_key_concept>`
 * :ref:`Consuming data <consuming_key_concept>`
 * :ref:`Saving data <saving_key_concept>`
+* :ref:`Streaming Execution Model <streaming_execution_model>`
 
-Datasets
---------
+.. _data_key_concepts:
+
+Key Concepts: Datasets and Blocks
+---------------------------------
+
+There are two main concepts in Ray Data:
+
+* :class:`Dataset <ray.data.Dataset>`
+* :ref:`Blocks <blocks_key_concept>`
+
+:class:`Dataset <ray.data.Dataset>` is the main user-facing Python API. It represents a
+distributed data collection and defines data loading and processing operations. You
+typically use the API in this way:
+
+1. Create a Dataset from external storage or in-memory data.
+2. Apply transformations to the data.
+3. Write the outputs to external storage or feed the outputs to training workers.
+
+The Dataset API is lazy, meaning that operations are not executed until you call an action
+like :meth:`~ray.data.Dataset.show`. This allows Ray Data to optimize the execution plan
+and execute operations in parallel.
+
+A *block* is the basic unit of data that Ray Data operates on. A block is a contiguous
+subset of rows from a dataset.
+
+The following figure visualizes a dataset with three blocks, each holding 1000 rows.
+Ray Data holds the :class:`~ray.data.Dataset` on the process that triggers execution
+(which is usually the driver) and stores the blocks as objects in Ray's shared-memory
```

Comment:
maybe link to driver definition in glossary? New ray user may not know exactly what it is